src/GuppyDataExplorer/ExplorerHeatMap/index.jsx

```diff
@@ -0,0 +1,185 @@
+          return res;
+        }, {});
+
+        yAxisVars.forEach((fieldName) => {
+          let rate;
+          if (fieldName === this.props.mainYAxisVar) {
+            const totalCount = this.props.filter[this.props.mainYAxisVar].selectedValues.length;
+            rate = details.count / totalCount;
+          } else {
+            rate = 1 - (fieldToMissingCount[fieldName] / details.count);
+          }
+          // Note: if '-' for zeros, there is no tooltip about which x/y is zero
+          rate = round(rate, 2); // 2 decimals / zero -> empty
+          transformedData.push([
+            xIndex,
+            yAxisVars.indexOf(fieldName),
+            rate, // round with 2 decimals in cells
+          ]);
+          if (rate > this.maxCellValue) {
+            // round UP with 1 decimal in legend
+            this.maxCellValue = Math.ceil(rate * precision) / precision;
+          }
+        });
+      });
+    }
+    return transformedData;
+  };
+
+  /**
+   * See echarts docs at https://echarts.apache.org/en/option.html
+   */
+  getHeatMapOptions = (data, xAxisVarTitle, yAxisVars, yAxisVarsMapping) => ({
+    tooltip: { // displayed when hover on cell
+      position: 'top',
+      formatter(params) {
+        // Note: params.data = [x, y, value]
+        const yField = yAxisVars[params.data[1]];
+        const mappingEntry = yAxisVarsMapping && yAxisVarsMapping.find(
+          i => i.field === yField,
+        );
+        const yAxisVar = (mappingEntry && mappingEntry.name) || yField;
+        return `Variable: ${capitalizeFirstLetter(yAxisVar)}<br/>${xAxisVarTitle}: ${params.data[0]}<br/>Data availability: ${params.data[2]}`;
+      },
+    },
+    grid: {
+      containLabel: true, // axis labels are cut off if not grid.containLabel
+      // distance to edges of the box:
+      top: 0,
+      bottom: 80, // space for x axis title and legend
+      left: '5%',
+      right: '5%',
+    },
+    xAxis: {
+      type: 'category',
+      axisLabel: {
+        show: false,
+      },
+      axisTick: {
+        show: false,
+      },
+      name: xAxisVarTitle,
+      nameLocation: 'middle',
+    },
+    yAxis: {
+      type: 'category',
+      data: yAxisVars.map((field) => {
+        const fieldMappingEntry = yAxisVarsMapping.find(i => i.field === field);
+        return capitalizeFirstLetter((fieldMappingEntry && fieldMappingEntry.name) || field);
+      }),
+      axisTick: {
+        show: false,
+      },
+      inverse: true,
+    },
+    visualMap: { // legend
+      min: 0,
+      max: this.maxCellValue, // round up (1 decimal) of max value
+      precision: 2, // 2 decimals in label
+      calculable: true, // handles on legend to adjust selected range
+      orient: 'horizontal',
+      left: 'center', // horizontal alignment
+      align: 'right', // position of bar relatively to handles and label
+      inRange: {
+        color: ['#EBF7FB', '#3188C6'], // [smallest value, greatest value]
```

Comment:
could you make these colors customizable? I'm thinking `['#EBF7FB', '#3188C6']` could be the default colors but if there is `categorical2Colors` defined in config files then use those two colors are heatmap color?

Something like this https://github.com/uc-cdis/data-portal/blob/master/src/GuppyDataExplorer/ExplorerVisualization/index.jsx#L67